### ISSUE TITLE
Resolve accessibility violations on various components

### ISSUE THREAD

**Issue by HollyJoyPhillips** [reapit/elements#53]
**Background:** 
The accessibility functionality has now been added to Elements (storybook) and gives us a better indication of where and if we need to update our components. For this ticket our main objective is listed under the violations tab and is indicated as either critical and or serious. 

This ticket is to review and update the associated violations on each component.  

**Task**: 
- Select a component in elements (list below)
- Click on the ‘Accessibility’ tab 
- Select the ‘Violations’ tab
- Resolve ‘critical’ items first 
- Resolve ‘serious’ items once any ‘critical’ items have been completed

**Example**
Component: Icon – Basic Usage: https://elements.reapit.cloud/?path=/story/icon--basic-usage

![Image](https://github.com/reapit/elements/assets/52408607/2df3c222-9d3e-47af-b801-24678b3be3f2)
 
**Issue**: 3 ‘serious’ violations

**To resolve:** 
- Add aria-label attribute 
- Add arial labelledby attribute
- Add title attribute

**Example attributes to obtain a pass:** 

![Image](https://github.com/reapit/elements/assets/52408607/ae12604c-2bf7-4657-90e0-46bf640c63e2)

**Note**: On each section the accessibility tab provides a link to related documentation and what is required to ensure the accessibility passes. 

**Components to review:** 

**Accordion**
- [x] Basic Usage
- [x] React Usage

**Related ticket:** https://github.com/reapit/elements/issues/79

**Avatar** 
- [x] Basic
- [x] With Src
- [x] With Image

**Badge** 
- [x] Basic Usage
- [x] With Intent 

**BreadCrumb**
- [x] Default Index

**Related ticket**: https://github.com/reapit/elements/issues/75

**Buttons**
- [x] Button Icons
- [x] Floating Button

**Related ticket:** https://github.com/reapit/elements/issues/83

**Card**
- [x] Card with Avatar
- [x] Card with Image
- [x] Card Focused
- [x] Card with List
- [x] Card Complete Example
- [x] React Shorthand Avatar & Body
- [x] React Shorthand List
- [x] React Shorthand Complete

**Chips**
- [x] Basic Usage
- [x] Chip Group Usage

**FileInput**
- [x] Default Usage
- [x] Advanced Usage

**FormLayout**
- [x] Basic Form
- [x] Complex Form

**Icon**
- [x] Basic Usage
- [x] Icon Font Size
- [x] Icon Custom Sizes
- [x] Icon Intent
- [x] Other Icons
- [x] Legacy Icons

**Input**
- [x] Input Number
- [x] Input Tel
- [x] Input URL
- [x] Input Search
- [x] Input Date
- [x] Input Time
- [x] Input Date-time Local
- [x] Input Week
- [x] Input Month
- [x] Input File
- [x] Input Checkbox
- [x] Input Radio

**Related ticket:** https://github.com/reapit/elements/issues/77 & https://github.com/reapit/elements/issues/80

**InputAddOn**
- [x] Basic Usage
- [x] With Icon
- [x] With Intent

**InputError**
- [x] Error Message

**InputGroup**
- [x] Complete Example
- [x] With Icon Only
- [x] With InputAddOn Only
- [x] React Shorthand
- [x] React Shorthand – custom ID
- [x] React Shorthand – all props 
- [x] Input checkbox
- [x] Input checkbox with Intent
- [x] Input checkbox No Label

**KeyValueList**
- [x] Basic Usage
- [x] Usage Large Grid

**Layouts**
- [x] Complete Combined Example
- [x] Complete Combined Example Max Width

**MobileControls**
- [x] Basic Usage

**Nav**
- [x] Basic example
- [x] Basic example Mobile Menu Open
- [x] React Usage
- [x] React Usage with Custom Brand 

**Related Ticket:** https://github.com/reapit/elements/issues/81 &https://github.com/reapit/elements/issues/82

**PageHeader**
- [x] Basic Usage
- [x] Pagination
- [x] Basic Example
- [x] React Example
- [x] React Example with Start End Buttons

**Pagination** 
- [x] Basic Example 
- [x] React Example
- [x] React Example with Start End Buttons

**Related ticket:** https://github.com/reapit/elements/issues/76

**PersistentNotification**
- [x] Full React example (fixed position)
- [x] Default Usage Inline 
- [x] With different icon
- [x] Inline notification, intent: primary
- [x] Inline notification: intent: neutral
- [x] Inline notification: intent: success
- [x] Inline notification: intent: pending
- [x] Inline notification: intent: warning
- [x] Inline notification: intent: danger
- [x] Inline notification: intent: default

**Related Ticket:** https://github.com/reapit/elements/issues/78

**PlaceholderImage**
- [x] Basic Usage
- [x] Fill Available Space

**ProgressBar**
- [x] React Example – ProgressBarPercentage
- [x] React Example – ProgressBarSteps

**SearchableDropdown**
- [x] Basic Usage
- [x] Controlled Usage

**Select**
- [x] Basic Usage

**Snack**
- [x] With an Icon
- [x] Intent: Primary
- [x] Intent: Neutral
- [x] Intent: Success
- [x] Intent: Pending
- [x] Intent: Warning
- [x] Intent: Danger
- [x] Intent: Default

**Table**
- [x] Basic Usage
- [x] Column Widths
- [x] Basic Customisation – Table Cells
- [x] Expanding Table Cell Space
- [x] Expandable Content
- [x] Call to Action Content
- [x] React Shorthand Usage
- [x] Advanced Customisation React

**Tabs**
- [x] Tabs – Default
- [x] Tabs – Full Width
- [x] Tabs – React Example uncontrolled
- [x] Tabs – React Example controlled

Related ticket: https://github.com/reapit/elements/issues/74

**TextArea**
- [x] Text Area Error

**Toggle**
- [x] Toggle - Grey Bg
- [x] Toggle Radio - Disabled

**Typography**
- [x] Tagged Typography
- [x] Typography Modifiers
- [x] Intent

Utilities 
- [x] Margin
- [x] Padding
- [x] Height and Width

For any areas you are uncertain or if you have any questions, please reach out to @HollyJoyPhillips or @willmcvay

**Comment by willmcvay** [comment]
Did what was possible on this - there are a number of violations that relate to colour contrast. Need to speak to the designers about this before re-visiting.
